krun.py

```diff
@@ -277,11 +281,34 @@ def inner_main(mailer, on_first_invocation, config, args):
         info(("Wait %s secs to allow system to cool prior to "
              "collecting initial temperature readings") %
              config.TEMP_READ_PAUSE)
+
+        # This part is wrapped in hooks, so that if daemons or networking are
+        # taken down for process executions, then the initial temperature
+        # reading gets the same treatment.
+        util.run_shell_cmd_list(config.PRE_EXECUTION_CMDS,)
         platform.sleep(config.TEMP_READ_PAUSE)
 
         debug("Taking fresh initial temperature readings")
         platform.starting_temperatures = platform.take_temperature_readings()
 
+        # Update the starting temperatures in the manifest
```

Comment:
Does this only happen on first invocation? (I think GH has not given us much context)